tests/test_grids.py

```diff
@@ -99,8 +99,63 @@ def test_avoid_repeated_grids():
     other_fields = {}
     other_fields['temp0'] = temp0_field
 
-    field_set2 = FieldSet(u_field, v_field, fields=other_fields)
-    assert field_set2.gridset.size == 2
-    assert field_set2.U.grid.name == 'grid0py'
-    assert field_set2.V.grid.name == 'grid1py'
-    assert field_set2.temp.grid.name == 'grid0py'
+    field_set = FieldSet(u_field, v_field, fields=other_fields)
+    assert field_set.gridset.size == 2
+    assert field_set.U.grid.name == 'grid0py'
+    assert field_set.V.grid.name == 'grid1py'
+    assert field_set.temp.grid.name == 'grid0py'
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('z4d', ['True', 'False'])
+def test_s_grids(mode, z4d):
```

Comment:
Is one unit test enough here? I would ideally also like to see some advection tests. I.e. particles moving in simple flow on (time-evolving?) s-grids